iOSClient/Activity/NCActivity.swift

```diff
@@ -351,69 +350,86 @@ extension NCActivity: UITableViewDataSource {
+    
+    func loadDataSource() {
+
+        var newItems = [DateCompareable]()
+        if showComments, let metadata = metadata, let account = NCManageDatabase.shared.getActiveAccount() {
+            let comments = NCManageDatabase.shared.getComments(account: account.account, objectId: metadata.fileId)
+            newItems += comments
+        }
+
         let activities = NCManageDatabase.shared.getActivity(
             predicate: NSPredicate(format: "account == %@", appDelegate.account),
             filterFileId: metadata?.fileId)
-        self.allItems += activities.filter
-        reloadDispatch.leave()
-        
-        reloadDispatch.notify(qos: .userInitiated, flags: .enforceQoS, queue: .main) {
-            self.allItems.sort(by: { $0.dateKey > $1.dateKey })
-            self.sectionDates = self.allItems.reduce(into: Set<Date>()) { partialResult, next in
-                   let newDay = Calendar.current.startOfDay(for: next.dateKey)
-                   partialResult.insert(newDay)
-            }.sorted(by: >)
-            self.tableView.reloadData()
-            NCUtility.shared.stopActivityIndicator()
-        }
+        newItems += activities.filter
+
+        self.allItems = newItems.sorted(by: { $0.dateKey > $1.dateKey })
+        self.sectionDates = self.allItems.reduce(into: Set<Date>()) { partialResult, next in
+            let newDay = Calendar.current.startOfDay(for: next.dateKey)
+            partialResult.insert(newDay)
+        }.sorted(by: >)
+        self.tableView.reloadData()
     }
     
-    @objc func loadActivity(idActivity: Int) {
-        
-        if !canFetchActivity { return }
-        canFetchActivity = false
-        
-        if idActivity > 0 {
-            
-            let height = self.tabBarController?.tabBar.frame.size.height ?? 0
-            NCUtility.shared.startActivityIndicator(backgroundView: self.view, blurEffect: false, bottom: height + 50, style: .gray)
+    func loadComments(disptachGroup: DispatchGroup? = nil) {
+        guard showComments, let metadata = metadata else { return }
+        disptachGroup?.enter()
+
+        NCCommunication.shared.getComments(fileId: metadata.fileId) { (account, comments, errorCode, errorDescription) in
+            if errorCode == 0, let comments = comments {
+                NCManageDatabase.shared.addComments(comments, account: metadata.account, objectId: metadata.fileId)
+            } else if errorCode != NCGlobal.shared.errorResourceNotFound {
+                NCContentPresenter.shared.messageNotification("_share_", description: errorDescription, delay: NCGlobal.shared.dismissAfterSecond, type: NCContentPresenter.messageType.error, errorCode: errorCode)
+            }
+
+            if let disptachGroup = disptachGroup {
+                disptachGroup.leave()
+            } else {
+                self.loadDataSource()
+            }
         }
-        
+    }
+    
+    func loadActivity(idActivity: Int, disptachGroup: DispatchGroup? = nil) {
```

Comment:
loadActivity is called always with disptachGroup(), so this arguments can't be optional (?)

Reply:
Correct 👍